### PR TITLE
Issue #2349 Improve counting of closed connections

### PR DESCRIPTION
**Pull request by gregw**
#2349 Improve closing stream counting by incrementing  on all paths through closing state machine.

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

**Comment by gregw** [comment]
@olamy this PR fails on my machine and on jenkins with out-of-memory???   It fails for all the client types (not just the http2, which I have changed).    Yet it passes for me in the main branch of jetty-9.4.x ???
Any idea what is going on?

**Comment by olamy** [comment]
@gregw  client.setMaxConnectionsPerDestination(Integer.MAX_VALUE); is the culprit.
This create huge empty collections in DuplexConnectionPool

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/AbstractTest.java [hunk 1]
```diff
@@ -206,6 +206,7 @@ protected void startClient() throws Exception
         clientThreads.setName("client");
         clientThreads.setDetailedDump(true);
         client = newHttpClient(provideClientTransport(transport), sslContextFactory);
+        client.setMaxConnectionsPerDestination(Integer.MAX_VALUE);
```
Comment (joakime) [inline tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/AbstractTest.java]: This allocates memory.
Set a more sane value here please.

Comment (gregw) [reply tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/AbstractTest.java]: Ah that would explain my out of memory exceptions!!! thank!

**Comment by gregw** [comment]
@joakime can you accept my changes after your review?
@sbordet what do you think of this one? shall we do a walk through?

**Comment by gregw** [comment]
Thanks @joakime , but @sbordet demolished the need for this in hangout review :)

**Pull request closed without merging**